### PR TITLE
Add file attachments to Mercury messages

### PR DESCRIPTION
**Pull request by mattnorris**
# Pull Request 

## Description

Add file attachments to Mercury messages. 

Fixes https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-66906

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jpjpjp** [comment]
Looks good!   I noticed that you are building a files field for EVERY message event and then have this code in the new getHydraFiles() method:
if (!files) {
    return undefined;
  }

This results in every message that DOESN'T have a file attachment getting having and undefined "files" attribute.

Would it be better to create the message event payload without files first and then add this logic before sending it:

if (activity.files) {
    messagesEventData.files = getHydraFiles(activity);
  }

File: packages/node_modules/@ciscospark/common/src/uuid-utils.js [hunk 1]
```diff
@@ -27,3 +31,29 @@ export function deconstructHydraId(id) {
...
+ */
+export function getHydraFiles(activity) {
+  const hydraFiles = [];
+  const {files} = activity.object;
+
+  if (files) {
+    const {items} = files;
+
+    // Note: Generated ID is dependent on file order.
+    for (let i = 0; i < items.length; i += 1) {
```
Comment (taymoork2) [inline packages/node_modules/@ciscospark/common/src/uuid-utils.js]: Could this be a `.forEach` or `.map` on the items array?

Comment (taymoork2) [reply packages/node_modules/@ciscospark/common/src/uuid-utils.js]: unless this is dependant on being as fast as possible

Comment (mattnorris) [reply packages/node_modules/@ciscospark/common/src/uuid-utils.js]: Unfortunately, no, it's dependent on position. Any other way has unused vars.

Comment (taymoork2) [reply packages/node_modules/@ciscospark/common/src/uuid-utils.js]: This is me being extra for the sake of being extra, but couldn't you do something like this?
```javascript
return activity.object.files
  ? activity.object.files.items.map(item =>
      hydraFiles.push(
        `${hydraBaseUrl}/contents/${constructHydraId(hydraTypes.CONTENT, `${activity.id}/${i}`)}`
      )
    )
  : [];
```

**Pull request merged**